src/openrct2/ride/SteamPosition.hpp

```diff
@@ -0,0 +1,99 @@
+        -49,  -97,  -165, -217,          // slopes up
+        49,   97,   165,  217,           // slopes down
+        -246, -256,                      // slopes vertical up
+        -247, -221, -181, -128, -66,  0, // slopes looping up
+        246,  256,                       // slopes vertical down
+        247,  221,  181,  128,  66,      // slopes looping down
+        -128, -221, -256, -221, -128,    // corkscrew up left
+        128,  221,  256,  221,  128,     // corkscrew down left
+        -128, -221, -256, -221, -128,    // corkscrew up right
+        128,  221,  256,  221,  128,     // corkscrew down right
+        -13,  -22,  13,   22,            // half helixes
+        -26,  26,                        // quarter helixes
+        -36,  -71,  -193,                // diagonal slopes up
+        36,   71,   193,                 // diagonal slopes down
+        97,   165,  217,                 // inverting transition slopes down
+        -44                              // spiral lift hill up
+    };
+
+    // COS((Y1/360)*2*PI())*256
+    // Where Y1 represents the angle of pitch in degrees
+    constexpr int16_t pitchCos[] = {
+        256,                                // Flat
+        251,  236,  195,  134,              // slopes up
+        251,  236,  195,  134,              // slopes down
+        70,   0,                            // slopes vertical up
+        -66,  -128, -181, -221, -247, -256, // slopes looping up
+        70,   0,                            // slopes vertical down
+        -66,  -128, -181, -221, -247,       // slopes looping down
+        221,  128,  0,    -128, -221,       // corkscrew up left
+        -221, -128, 0,    128,  221,        // corkscrew down left
+        221,  128,  0,    -128, -221,       // corkscrew up right
+        -221, -128, 0,    128,  221,        // corkscrew down right
+        255,  255,  255,  255,              // half helixes
+        254,  254,                          // quarter helixes
+        253,  245,  168,                    // diagonal slopes up
+        253,  245,  168,                    // diagonal slopes down
+        236,  195,  134,                    // slopes up
+        252                                 // spiral lift hill up
+    };
+
+    constexpr auto computeXYMagnitude(int32_t height, uint8_t pitch)
+    {
+        return (pitchSin[static_cast<uint8_t>(pitch)] * height) / 256;
+    }
+
+    constexpr CoordsXY computeXYVector(int32_t magnitude, uint8_t yaw)
+    {
+        return (yawToDirectionVector[yaw] * magnitude) / 256;
+    }
+    constexpr CoordsXY computeXYVector(int32_t height, uint8_t pitch, uint8_t yaw)
+    {
+        return computeXYVector(computeXYMagnitude(height, pitch), yaw);
+    }
+    constexpr CoordsXYZ computeXYZVector(int32_t height, int32_t run, uint8_t pitch, uint8_t yaw)
+    {
+        int32_t sin = pitchSin[static_cast<uint8_t>(pitch)];
+        int32_t cos = pitchCos[static_cast<uint8_t>(pitch)];
```

Comment:
Avoid using sin and cos although not protected words they are in the c api so may cause name collisions.